gdnative-core/src/class.rs

```diff
@@ -288,6 +375,36 @@ class $name:ident: $owner:ty {
             }
         }
     );
+    (
+class $name:ident: $owner:ty {
+    fields {
+        $(
+            $(#[$fattr:meta])*
+            $fname:ident : $fty:ty,
+        )*
+    }
+    setup($builder:ident) $pbody:block
+    constructor($owner_name:ident : $owner_ty:ty) $construct:block
+
+    $($tt:tt)*
+}
+    ) => (
+        godot_class! {
+            class $name ($crate::user_data::MutexData<$name>) : $owner {
```

Comment:
This is just personal preference, but I would probably prefer to have the default be the single-threaded version (so `RefCell` based). Most scripts will not be used in multiple threads so this might be more costly than it is useful in the general case.

Reply:
(After finishing reviewing I realized single-threaded versions don't exist, so feel free to ignore this)

Reply:
It is true that `_physics_process` and `_process` happen on the same thread, but signals connections seem to get called immediately unless they are `CONNECT_DEFERRED`. That's quite some chance to get exposed to threads, imo.

`parking_lot` mutexes need only one atomic operation for uncontended locks, so it may not be as scary as it sounds, but a feature flag should be useful, as said in replies to other comments on this topic.